File: Source/Padutronics.Windows.Win32.Api/D3D11/ID3D11Device.cs
using Padutronics.Windows.Win32.Api.Unknwn;
using System.Runtime.InteropServices;

namespace Padutronics.Windows.Win32.Api.D3D11;

[Guid("db6f6ddb-ac77-4e88-8253-819df9bbf140")]
public interface ID3D11Device : IUnknown
{
    void CreateRenderTargetView(ID3D11Resource pResource, D3D11_RENDER_TARGET_VIEW_DESC? pDesc, out ID3D11RenderTargetView ppRTView);
    void GetImmediateContext(out ID3D11DeviceContext ppImmediateContext);
}

File: Source/Padutronics.Windows.Win32.Api.CWrapper/D3D11/ID3D11Device.cpp
#include "ID3D11Device.hpp"

auto ID3D11Device_CreateRenderTargetView(ID3D11Device& _this, ID3D11Resource* pResource, const D3D11_RENDER_TARGET_VIEW_DESC* pDesc, ID3D11RenderTargetView** ppRTView) -> HRESULT
{
    return _this.CreateRenderTargetView(pResource, pDesc, ppRTView);
}

void ID3D11Device_GetImmediateContext(ID3D11Device& _this, ID3D11DeviceContext** ppImmediateContext)
{
    _this.GetImmediateContext(ppImmediateContext);
}